service/paypal.go

```diff
@@ -0,0 +1,112 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/companieshouse/payments.api.ch.gov.uk/config"
+	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/plutov/paypal/v4"
+)
+
+var client *paypal.Client
+
+func getPayPalClient(cfg config.Config) (*paypal.Client, error) {
+	if client != nil {
+		return client, nil
+	}
+
+	paypalAPIBase := getPayPalAPIBase(cfg.PaypalEnv)
+	if paypalAPIBase == "" {
+		return nil, fmt.Errorf("invalid paypal env in config: %s", cfg.PaypalEnv)
+	}
+
+	c, err := paypal.NewClient(cfg.PaypalClientID, cfg.PaypalSecret, paypalAPIBase)
+	if err != nil {
+		return nil, fmt.Errorf("error creating paypal client: [%v]", err)
+	}
+	_, err = c.GetAccessToken(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("error getting access token: [%v]", err)
+	}
+
+	return c, nil
+}
+
+// PayPalSDK is an interface for all the PayPal client methods that will be used
+// in this service
+type PayPalSDK interface {
+	GetAccessToken(ctx context.Context) (*paypal.TokenResponse, error)
+	CreateOrder(ctx context.Context, intent string, purchaseUnits []paypal.PurchaseUnitRequest, payer *paypal.CreateOrderPayer, appContext *paypal.ApplicationContext) (*paypal.Order, error)
+}
+
+// PayPalPaymentProvider is an interface to enable mocking
+type PayPalPaymentProvider interface {
+	CreatePayPalOrder(paymentResource *models.PaymentResourceRest) (string, ResponseType, error)
+}
+
+// PayPalService handles the specific functionality of integrating PayPal into Payment Sessions
+type PayPalService struct {
+	Client         PayPalSDK
+	PaymentService PaymentService
+}
+
+func NewPayPalService(cfg config.Config, paymentSvc PaymentService) (*PayPalService, error) {
+	c, err := getPayPalClient(cfg)
```

Comment:
You can pass in `paymentSvc.Config` here to get the config and not have to pass it is as a parameter